src/test/lib/ApiTest.cpp

```diff
@@ -6530,3 +6627,225 @@ QuicTestValidateConnectionPoolCreate()
     }
 }
 #endif // QUIC_API_ENABLE_PREVIEW_FEATURES
+
+void
+QuicTestRetryConfigSetting()
+{
+    uint8_t TestSecret[32]{};
+    uint8_t ResultBuffer[sizeof(QUIC_STATELESS_RETRY_CONFIG) + 32]{};
+    QUIC_STATELESS_RETRY_CONFIG* ResultConfig = (QUIC_STATELESS_RETRY_CONFIG*)ResultBuffer;
+    uint32_t ResultBufferSize = sizeof(ResultBuffer);
+    uint32_t TestRotationMs = 54321;
+    QUIC_AEAD_ALGORITHM_TYPE TestAlg = QUIC_AEAD_ALGORITHM_AES_128_GCM;
+    MsQuicRegistration Registration("TestRetryConfigSetting");
+    {
+        QuicStorageSettingScopeGuard GlobalSettings = QuicStorageSettingScopeGuard::Create();
+
+        //
+        // Test that the defaults were correctly picked up.
+        //
+        TEST_QUIC_SUCCEEDED(
+            MsQuic->GetParam(
+                nullptr,
+                QUIC_PARAM_GLOBAL_STATELESS_RETRY_CONFIG,
+                &ResultBufferSize,
+                ResultBuffer));
+        TEST_EQUAL(QUIC_AEAD_ALGORITHM_AES_256_GCM, ResultConfig->Algorithm);
+        TEST_EQUAL(32, ResultConfig->SecretLength);
+        TEST_EQUAL(QUIC_STATELESS_RETRY_KEY_LIFETIME_MS, ResultConfig->RotationMs);
+
+        //
+        // Test Key Rotation Ms is picked up automatically
+        //
+        TEST_QUIC_SUCCEEDED(
+            CxPlatStorageWriteValue(
+                GlobalSettings,
+                QUIC_SETTING_RETRY_KEY_ROTATION_MS,
+                CXPLAT_STORAGE_TYPE_UINT32,
+                sizeof(TestRotationMs),
+                (uint8_t*)&TestRotationMs));
+
+        CxPlatSleep(100);
```

Comment:
This makes me affraid of test flakyness.
In similar situations, I have used a "TryUntil" wrapper, that read in a loop with a short sleep time (100ms), re-try until a condition is acheived, or that a long timeout (2s) elapsed.

It is pretty readable in C++ (make the condition a lambda) and you get the best of:
- a fast test execution
- no flakiness

Reply:
I agree that you might want retry here. Also 100 ms might not be enough based on prior experience.

Reply:
I've implemented this because I agree with the utility, but not all of these test cases work in this manner. Some of them test that nothing has changed, so there's nothing to poll for, but we still need to wait until some time has passed for the registry notification to fire and be processed.  I'm not sure there's something I can poll for toe determine if the registry has been read and processed in that instance.